src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponent.scala

```diff
@@ -24,12 +29,114 @@ trait AttributeComponent {
     def name = column[String]("name")
     def valueString = column[Option[String]]("value_string")
     def valueNumber = column[Option[Double]]("value_number")
-    def valueDate = column[Option[Timestamp]]("value_date")
     def valueBoolean = column[Option[Boolean]]("value_boolean")
+    def valueEntityRef = column[Option[Long]]("value_entity_ref")
     def listIndex = column[Option[Int]]("list_index")
 
-    def * = (id, name, valueString, valueNumber, valueDate, valueBoolean, listIndex) <> (AttributeRecord.tupled, AttributeRecord.unapply)
+    def * = (id, name, valueString, valueNumber, valueBoolean, valueEntityRef, listIndex) <> (AttributeRecord.tupled, AttributeRecord.unapply)
+
+    def workspace = foreignKey("FK_ATTRIBUTE_ENTITY_REF", valueEntityRef, entityQuery)(_.id.?)
   }
 
-  protected val attributeQuery = TableQuery[AttributeTable]
+  protected object attributeQuery extends TableQuery(new AttributeTable(_)) {
```

Comment:
I like this structure of extending TableQuery (as mentioned [here](http://slick.typesafe.com/doc/3.1.1/schemas.html#table-query) ). This gives us a namespace for our functions so our data access component won't have 10 overloaded save functions.